src/coreComponents/linearAlgebra/DofManager.cpp

```diff
@@ -456,31 +474,60 @@ void DofManager::addCoupling( string const & rowFieldName,
                               std::vector< Regions > const & regions,
                               bool const symmetric )
 {
-  GEOSX_ASSERT_MSG( m_domain != nullptr, "Mesh has not been set" );
+  GEOSX_ASSERT_MSG( m_domain != nullptr, "Domain has not been set" );
   localIndex const rowFieldIndex = getFieldIndex( rowFieldName );
   localIndex const colFieldIndex = getFieldIndex( colFieldName );
 
   // Check if already defined
-  GEOSX_ASSERT_MSG( m_coupling.count( {rowFieldIndex, colFieldIndex} ) == 0, "addCoupling: coupling already defined" );
+  std::vector< Regions > processedRegionList = processCouplingRegionList( regions,
+                                                                          m_fields[rowFieldIndex].support,
+                                                                          rowFieldName,
+                                                                          m_fields[colFieldIndex].support,
+                                                                          colFieldName );
 
-  CouplingDescription & coupling = m_coupling[ { rowFieldIndex, colFieldIndex } ];
-  coupling.connector = connectivity;
-  if( connectivity == Connector::None && rowFieldIndex == colFieldIndex )
+  if( m_coupling.count( {rowFieldIndex, colFieldIndex} ) == 0 )
   {
-    coupling.connector = static_cast< Connector >( m_fields[rowFieldIndex].location );
-  }
+    CouplingDescription & coupling = m_coupling[ { rowFieldIndex, colFieldIndex } ];
+    coupling.connector = connectivity;
+    if( connectivity == Connector::None && rowFieldIndex == colFieldIndex )
+    {
+      coupling.connector = static_cast< Connector >( m_fields[rowFieldIndex].location );
+    }
 
-  // Make a list of regions on which coupling is defined
-  coupling.support = processCouplingRegionList( regions,
-                                                m_fields[rowFieldIndex].support,
-                                                rowFieldName,
-                                                m_fields[colFieldIndex].support,
-                                                colFieldName );
+    // Make a list of regions on which coupling is defined
+    coupling.support = processedRegionList;
 
-  // Set connectivity with active symmetry flag
-  if( symmetric && colFieldIndex != rowFieldIndex )
+    // Set connectivity with active symmetry flag
+    if( symmetric && colFieldIndex != rowFieldIndex )
+    {
+      m_coupling.insert( { { colFieldIndex, rowFieldIndex }, coupling } );
+    }
+  }
+  else
   {
-    m_coupling.insert( { { colFieldIndex, rowFieldIndex }, coupling } );
+    CouplingDescription & coupling = m_coupling[ { rowFieldIndex, colFieldIndex } ];
+
+    for( auto const & newRegion : processedRegionList )
+    {
+      auto it    = coupling.support.begin();
+      auto end1  = coupling.support.end();
+      bool added = false;
+      while( it != end1 && !added )
+      {
+        added = it->add( newRegion );
+        it++;
+      }
```

Comment:
```suggestion
      bool added = false;
      for( Regions & regions: coupling.support )
      {
        added = regions.add( newRegion );
        if( added ) break;
      }
```
Would this be equivalent?

Reply:
looks equivalent to me.